packages/cli/lib/SpinniesManager.js

```diff
@@ -0,0 +1,42 @@
+const Spinnies = require('spinnies');
+
+// Allows us to maintain a single instance of spinnies across multiple files
+class SpinniesManager {
```

Comment:
The comment explains it, but this allows us to maintain a singe instance of Spinnies without having to pass the instance around as function arguments. This is useful because we have util files that instantiate and use Spinnies that can conflict with the command-level spinnies instances. 

I also overrode the `add` function to allow for a special option `isParent`. This lets us automatically nest future Spinnies lines under a parent. The spinnies api is pretty simple (only like 8 available functions to call), so it feels like it's ok to make this abstraction. It will also allow us to override other functions down the road if we want more control.

The requirement to use this new wrapper is to instantiate Spinnies with `SpinniesManager.init(options)` rather than `new Spinnies(options)`. I didn't update all the usages of spinnies with this new init pattern, but I think we should be able to safely do that later if we want to.